src/components/dropdown/dropdown.tsx

```diff
@@ -439,14 +448,39 @@ export class Dropdown implements InteractiveComponent {
   setScrollerEl = (scrollerEl: HTMLDivElement): void => {
     this.resizeObserver.observe(scrollerEl);
     this.scrollerEl = scrollerEl;
+    this.scrollerEl.addEventListener("transitionrun", this.onTransitionRun);
   };
 
   transitionEnd = (event: TransitionEvent): void => {
     if (event.propertyName === this.activeTransitionProp) {
-      this.active ? this.calciteDropdownOpen.emit() : this.calciteDropdownClose.emit();
+      this.active ? this.openCloseEventEmitter("open") : this.openCloseEventEmitter("close");
     }
   };
 
+  onTransitionRun = (event: TransitionEvent): void => {
+    if (event.propertyName === this.activeTransitionProp) {
+      this.active
+        ? this.openCloseEventEmitter("beforeOpen")
+        : this.openCloseEventEmitter("beforeClose");
+    }
+  };
```

Comment:
I would recommend renaming this to `transitionRunHandler` to be more explicit that it is a handler function as opposed to an event.